### PR TITLE
[doc] update user guide 6.8

### PR DESCRIPTION
**Pull request by Rhett-Ying**
## Checklist
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [$CATEGORY] (such as [NN], [Model], [Doc], [Feature]])
- [ ] I've leverage the [tools](https://docs.google.com/document/d/1iHyj7zlmygKSk5gBPsqIqL5ASPzJSPREaNT_QdsiYA4/edit) to beautify the python and c++ code.
- [ ] The PR is complete and small, read the [Google eng practice (CL equals to PR)](https://google.github.io/eng-practices/review/developer/small-cls.html) to understand more about small PR. In DGL, we consider PRs with less than 200 lines of core code change are small (example, test and documentation could be exempted).
- [ ] All changes have test coverage
- [ ] Code is well-documented
- [ ] To the best of my knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change
- [ ] Related issue is referred in this PR
- [ ] If the PR is for a new model/paper, I've updated the example index [here](../examples/README.md).

### CODE REVIEW

File: docs/source/guide/minibatch-prefetching.rst [hunk 1]
```diff
@@ -3,94 +3,25 @@
 6.8 Feature Prefetching
```
Comment (frozenbugs) [inline docs/source/guide/minibatch-prefetching.rst:3]: Consider change the title entirely, e.g. datapipe overlapping, and rewrite it to illustrate how we handles overlapping different stage with the advantage of new graphbolt datapipe mechanism.

Comment (mfbalin) [reply docs/source/guide/minibatch-prefetching.rst:3]: Shall we use "pipeline parallelism" or "pipelining" instead of overlapping or prefetching? It would fit the GPU graph sampling pipelining approach in our proposal better.

**Review by frozenbugs — APPROVED** [review]

File: docs/source/guide/minibatch-parallelism.rst [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+`IterableDataPipe <https://pytorch.org/data/main/torchdata.datapipes.iter.html>`__
+and stacked together with `PyTorch DataLoader <https://pytorch.org/docs/stable/data
+.html#torch.utils.data.DataLoader>`__.
+This design allows us to easily customize the data loading process by
+chaining different data pipes together. For example, if we want to sample
+negative items for each seed from graph, we can simply chain the
+:class:`~dgl.graphbolt.NegativeSampler` after the :class:`~dgl.graphbolt.ItemSampler`.
+
+But simply chaining data pipes together incurs performance overheads. Considering
+the fact that some stages are computationally intensive while others are I/O intensive,
```
Comment (frozenbugs) [inline docs/source/guide/minibatch-parallelism.rst:38]: instead mentioning computationally intensive and I/O intensive, probably better to say they are using different hardware resources which includes CPU intensive, GPU intensive, PCIe (CPU to GPU), Disk e.t.c.

Comment (Rhett-Ying) [reply docs/source/guide/minibatch-parallelism.rst]: updated.
